### ISSUE TITLE
(Big Sur) AdGuard blocks git pushes via ssh

### ISSUE THREAD

**Issue by akmLaVx** [AdguardTeam/AdguardForMac#735]
### Steps to reproduce
1. Launch AdGuard
2. Try a git push through ssh
3. You will receive the following error:

```
> git push kex_exchange_identification: write: Broken pipe
fatal: Could not read from remote repository
```

### Expected behavior
AdGuard doesn't touch Git

### Actual behavior
AdGuard borks Git

### Customer ID & logs
See helpdesk ID 2884985.

### Your environment
* Environment name and version: macOS Big Sur (10.16, Build 20A5354i), AdGuard 2.5.0.849n

**Comment by HelgeBecker** [comment]
Confirmed.

**Comment by Chinaski1** [comment]
@HelgeBecker 

Hello!
Please, update to the [Nightly](agrd.io/mac_nightly) version of the Adguard app.
Is the issue still reproduce?

**Comment by akmLaVx** [comment]
@Chinaski1 User was already on Nightly.

**Comment by Chinaski1** [comment]
I mean the latest nightly version 2.5.0.864 nightly (CL-1.7.85). 
Judging by the logos he uses 2.5.0.849.

**Comment by bermudalocket** [comment]
This seems to occur with any key-based network interactions. I cannot SSH into any remote servers or push to GitHub with AdGuard enabled. There is nothing in the filtering log even with logging level set to debug.

This is an absolute show-stopper and hope this can be fixed soon.

Error:

    kex_exchange_identification: write: Broken pipe

Version:

    Application version 2.5.0.864 nightly (CL-1.7.85)

OS:

    macOS 11.0 Beta (20A5364e)

**Comment by sfionov** [comment]
@bermudalocket Can you please try with openssh from Homebrew? Seems that Apple is porting its system software to its new network framework that doesn't work well when network extension is active.

**Comment by bermudalocket** [comment]
@sfionov I can confirm that ssh works as expected after installing OpenSSH 8.3p1 from Homebrew. Thank you!!

**Comment by glowacki-dev** [comment]
I had this issue before, but it seems to be fixed with the latest build: AdGuard 2.5.0.879 nightly, macos 11.0 Beta (20A5364e). I have no more issues using git over ssh with adguard enabled.

I believe I'm using openssh provided by Apple, so there was no need to install in from homebrew
```
$ ssh -V
OpenSSH_8.1p1, LibreSSL 2.7.3
```

**Comment by Chinaski1** [comment]
Fixed on AdGuard 2.5.0.879 nightly
